crates/client-api/src/auth.rs

```diff
@@ -260,3 +243,15 @@ impl headers::Header for SpacetimeExecutionDurationMicros {
         values.extend([(self.0.as_micros() as u64).into()])
     }
 }
+
+pub async fn auth_middleware<S: ControlStateDelegate + NodeDelegate>(
```

Comment:
what do you think about renaming this to something more descriptive? e.g. `anonymous_auth_middleware`?

Reply:
Not fully opposed, but I think this is descriptive enough as is - it's a feature of our auth system that you don't actually *need* to be auth'd, that we'll generate an identity for you. So this does do standard authentication, not just anonymous auth.

Reply:
The only places we don't generate it for you is where it doesn't make any sense to - but there's no reason to disallow it, just that it doesn't make much sense.